apps/openmw/mwmechanics/aipackage.cpp

```diff
@@ -179,10 +180,33 @@ void MWMechanics::AiPackage::evadeObstacles(const MWWorld::Ptr& actor, float dur
     if (door != MWWorld::Ptr())
     {
         // note: AiWander currently does not open doors
-        if (getTypeId() != TypeIdWander && !door.getCellRef().getTeleport() && door.getCellRef().getTrap().empty()
-                && door.getCellRef().getLockLevel() <= 0 && door.getClass().getDoorState(door) == 0)
+        if (getTypeId() != TypeIdWander && !door.getCellRef().getTeleport() && door.getClass().getDoorState(door) == 0)
         {
-            MWBase::Environment::get().getWorld()->activateDoor(door, 1);
+            if ((door.getCellRef().getTrap().empty() && door.getCellRef().getLockLevel() <= 0 ))
+                MWBase::Environment::get().getWorld()->activate(door, actor);
```

Comment:
Looks like you call activate() twice for a door that needs a key but is already unlocked?